pkg/operator/configobservation/images/observe_images.go

```diff
@@ -56,3 +59,117 @@ func ObserveInternalRegistryHostname(genericListers configobserver.Listers, reco
 	}
 	return observedConfig, errs
 }
+
+func ObserveExternalRegistryHostnames(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+	listers := genericListers.(configobservation.Listers)
+	var errs []error
+	prevObservedConfig := map[string]interface{}{}
+
+	// first observe all the existing config values so that if we get any errors
+	// we can at least return those.
+	externalRegistryHostnamePath := []string{"imagePolicyConfig", "externalRegistryHostnames"}
+	o, _, _ := unstructured.NestedSlice(existingConfig, externalRegistryHostnamePath...)
+	if len(o) > 0 {
+		err := unstructured.SetNestedSlice(prevObservedConfig, o, externalRegistryHostnamePath...)
+		if err != nil {
+			return prevObservedConfig, append(errs, err)
+		}
+	}
+
+	if !listers.ImageConfigSynced() {
+		glog.Warning("images.config.openshift.io not synced")
```

Comment:
@deads2k @sttts we already log this messages in existing function, this will double the amount of log messages I guess?

Reply:
This logging inside of config observers does not make much sense IMO. We should switch to a condition which shows a more detailed error, i.e. "imagePolicyConfig.externalRegistryHostnames in images.config.openshift.io not found".

Reply:
i'm not sure how that is a more detailed error.  Or even related to the condition we are checking for here.

Reply:
@bparees i think @sttts meant to add new condition instead of emitting a lot of warning logs into stderr. Something like 'WaitingForImageConfig=true/false' or something similar? Does not have to be this PR, can be a follow up.